tests/lazy-init.html

```diff
@@ -0,0 +1,111 @@
+<!doctype html>
+<!--
+@license
+Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+<script>WCT = { waitFor(cb) { addEventListener('DOMContentLoaded', cb) } };</script>
+<script src="test-flags.js"></script>
+<script src="../node_modules/wct-browser-legacy/browser.js"></script>
+<script src="../node_modules/@webcomponents/webcomponents-platform/webcomponents-platform.js"></script>
+<script src="../node_modules/es6-promise/dist/es6-promise.auto.min.js"></script>
+<script src="../node_modules/@webcomponents/template/template.js"></script>
+<script src="../node_modules/@webcomponents/html-imports/html-imports.min.js"></script>
+<script src="../node_modules/@webcomponents/shadydom/shadydom.min.js"></script>
+<script src="../node_modules/@webcomponents/custom-elements/custom-elements.min.js"></script>
+<script src="../scoping-shim.min.js"></script>
+<script src="../apply-shim.min.js"></script>
+<script src="../custom-style-interface.min.js"></script>
+<template id="x-foo">
+  <style>
+    :host {
+      display: block;
+      height: 100px;
+      background-color: blue;
+    }
+
+    :host > x-bar {
+      --foo: rgb(255, 0, 0);
+    }
+  </style>
+  <x-bar></x-bar>
+</template>
+<template id="x-bar">
+  <style>
+    :host {
+      display: block;
+      color: var(--foo);
+    }
+
+    div {
+      border: 2px solid rgb(0, 255, 0);
+      border-color: var(--foo);
+    }
+  </style>
+  <div>Hello!</div>
+</template>
+<script>
+  class LazyElement extends HTMLElement {
+    constructor() {
+      super();
+      this.initialized = false;
+      this.attachShadow({mode: 'open'});
+    }
+    init() {
+      if (this.initialized) {
+        return;
+      }
+      this.initialized = true;
+      const template = document.querySelector(`template#${this.localName}`);
+      if (!template.initialized) {
```

Comment:
Is this condition needed?

Reply:
This would only be necessary if there were multiple instances of this element.